edx_adapt/data/course_repository.py

```diff
@@ -516,6 +236,28 @@ def advance_problem(self, course_id, user_id):
         current_problem = self.store.get_one(coll, user_id, 'next')
         self.store.update_doc(coll, {'student_id': user_id}, {'$set': {'current': current_problem, 'next': None}})
 
+    def get_all_remaining_problems(self, course_id, user_id):
+        return [x for x in self._get_remaining_by_user(course_id, user_id)
+                if x['pretest'] is False and x['posttest'] is False]
+
+    def get_remaining_problems(self, course_id, skill_name, user_id):
+        remaining = self.get_all_remaining_problems(course_id, user_id)
+        return [x for x in remaining if skill_name in x['skills']]
+
+    def get_all_remaining_posttest_problems(self, course_id, user_id):
```

Comment:
It's not quite clear what's the difference between this method and `get_remaining_posttest_problems`?